srcs/Socket.cpp

```diff
@@ -62,7 +65,38 @@ void	Socket::printConnection(void) {
 	std::cout << "server received connection from: " << s << std::endl;
 }
 
-
+void	Socket::initListenSocket(void) {
+	struct addrinfo	hints;
+	struct addrinfo	*ai;
+	struct addrinfo *p;
+
+	// Get a socket for us and bind it
+	memset(&hints, 0, sizeof(hints)); // change to ft_memset later?
+	hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;
+	if (getaddrinfo(NULL, PORT, &hints, &ai) != 0)
+		throw SocketInitException();
+	for (p = ai; p != NULL; p = p->ai_next)
+	{
+		try {
+			createSocket(p->ai_family, p->ai_socktype, p->ai_protocol);
+		} catch (std::exception& e) {
```

Comment:
I would write the error to ease the debug if something bad occurs

Reply:
The error is defined inside of the createSocket method:

```
void	Socket::createSocket(int ai_family, int ai_socktype, int ai_protocol) {
	int yes = 1;

	if ((sockfd = socket(ai_family, ai_socktype, ai_protocol)) < 0)
		throw Socket::SocketFailureException();
	if (setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(int)) < 0)
		throw Socket::SocketSetOptException();
	std::cout << "listenSocket set up successfully" << std::endl;
}
```

Or did you mean something else?

Reply:
I see that you throw an exception but then, when the exception gets caught I would do something like 

```c
... } catch (std::exception& e) {
  std::cerr << e.what() << std::endl; 
  continue ;
}
```

https://learn.microsoft.com/en-us/cpp/cpp/errors-and-exception-handling-modern-cpp?view=msvc-170 here they suggest more or less the same. There might be a reason why you have not written it - if that's the case, please, tell me :)